packages/@aws-cdk/aws-iam/lib/policy-statement.ts

```diff
@@ -531,6 +531,44 @@ export interface PolicyStatementProps {
   readonly effect?: Effect;
 }
 
+interface PolicyStatementActions extends PolicyStatementBase {
+  /**
+   * List of actions to add to the statement
+   *
+   * @default - no actions
+   */
+  readonly actions: string[];
+}
+
+interface PolicyStatementNotActions extends PolicyStatementBase {
+  /**
+   * List of not actions to add to the statement
+   *
+   * @default - no not-actions
+   */
+  readonly notActions: string[];
+}
+
+interface PolicyStatementResources extends PolicyStatementBase {
+  /**
+   * Resource ARNs to add to the statement
+   *
+   * @default - no resources
+   */
+  readonly resources: string[];
+}
+
+interface PolicyStatementNotResources extends PolicyStatementBase {
+  /**
+   * NotResource ARNs to add to the statement
+   *
+   * @default - no not-resources
+   */
+  readonly notResources: string[];
+}
+
+export type PolicyStatementProps =  (PolicyStatementActions | PolicyStatementNotActions) & (PolicyStatementResources | PolicyStatementNotResources);
```

Comment:
We can't support type declarations like this in jsii.

Your build should have yelled at you.